src/modules/_networking/private_dns_zone/_locals.tf

```diff
@@ -0,0 +1,29 @@
+locals {
+  resource_group = var.resources.resource_groups[var.settings.resource_group_ref]
+
+  resource_group_name = local.resource_group.name
+  location            = local.resource_group.location
+  tags = merge(
+    var.global_settings.tags,
+    var.global_settings.inherit_resource_group_tags ? local.resource_group.tags : {},
+    try(var.settings.tags, {})
+  )
+  vnet_ids = {
+    for vnet in var.settings.vnet_ref :
+    vnet => {
+      name = var.resources.virtual_networks[vnet].name
+      id   = var.resources.virtual_networks[vnet].id
+    }
+  }
+}
+locals {
+  # local object used to map possible private dns zoone names
+  zone_names = {
+    "storage_blob"   = "privatelink.blob.core.windows.net"
```

Comment:
`storage_blob` is the only name in this map that is not plural. Consistency is key. Reconsider.